### PR TITLE
Add test: Should Implement service.kubernetes.io/service-proxy-name

### PR DESCRIPTION
**Pull request by HanFa**
E2E tests for https://github.com/kubernetes/kubernetes/blob/da707b6133ff0c66dc4d24295374e7e5de8852f5/test/e2e/network/service.go#L1889

```
2021-11-28T03:00:12.536-0800	info	Tests passed, validation succeeded!
--- PASS: TestProxyName (124.79s)
    --- PASS: TestProxyName/ProxyName (124.79s)
        --- PASS: TestProxyName/ProxyName/should_implement_service.kubernetes.io/service-proxy-name (123.29s)
```

### CODE REVIEW

**Review by knabben — DISMISSED** [review]
🚀

File: tests/proxyname_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+func servicesRemoveLabel(services kubernetes.Services, key string) error {
+	for _, service := range services {
+		if err := service.RemoveLabel(key); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func TestProxyName(t *testing.T) {
```
Comment (knabben) [inline tests/proxyname_test.go:56]: Can we rename to something closer like `s/ProxyName/ProxyNamelabel`, so we kind of know this is related to a label and not a feature...

Comment (HanFa) [reply tests/proxyname_test.go:56]: yea sure!

File: tests/proxyname_test.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+	var disabledClusterIPs []string
+	var toggledClusterIPs []string
+
+	labelKey := "service.kubernetes.io/service-proxy-name"
+	labelValue := "foo-bar"
+
+	featureProxyName := features.New("ProxyName").WithLabel("type", "proxyName").
+		Setup(func(_ context.Context, t *testing.T, _ *envconf.Config) context.Context {
+			for _, pod := range pods {
```
Comment (HanFa) [inline tests/proxyname_test.go:69]: do we need services for 1 or 4 pods?

Comment (HanFa) [reply tests/proxyname_test.go:69]: currently, we have 2 svcs for each pod, 8 in total

option: we can have one pod (having two svcs on it) as target and other pods try to connect to it

Comment (yzaccc) [inline tests/proxyname_test.go:69]: Do we need to create 2 services for each pod, totally 8 services?
I think we may just need to create services on one pod.

Comment (HanFa) [reply tests/proxyname_test.go:69]: use option

Comment (yzaccc) [reply tests/proxyname_test.go:69]: what did you change?

Comment (HanFa) [reply tests/proxyname_test.go]: changed to only 2 services, deployed on the first pods

**Review by yzaccc — APPROVED** [review]
LGTM

**Comment by yzaccc** [comment]
CI pipeline filed by other issues, this PR is good to squash and merge.

**Pull request merged**